From 94cf62bdf4a9e74ec4613f3df8ae1e7e925e95ba Mon Sep 17 00:00:00 2001
From: Jun Yuan Tan <junyuan.tan@starfivetech.com>
Date: Tue, 9 Nov 2021 10:34:47 +0800
Subject: [PATCH 18/34] fix path to libffi

FFI_LIBRARY_PATH is the full path to libffi so when building something that links to
libLLVMInterpreter, we were looking for libffi in clang's recipe-sysroot instead of
the recipe's sysroot.

Rebased to LLVM 14.0.0 by Jun Yuan Tan

Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Jun Yuan Tan <junyuan.tan@starfivetech.com>
---
 llvm/lib/ExecutionEngine/Interpreter/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/lib/ExecutionEngine/Interpreter/CMakeLists.txt b/llvm/lib/ExecutionEngine/Interpreter/CMakeLists.txt
index 976219830d5e..9930e01559e9 100644
--- a/llvm/lib/ExecutionEngine/Interpreter/CMakeLists.txt
+++ b/llvm/lib/ExecutionEngine/Interpreter/CMakeLists.txt
@@ -22,5 +22,5 @@ add_llvm_component_library(LLVMInterpreter
   )
 
 if( LLVM_ENABLE_FFI )
-  target_link_libraries( LLVMInterpreter PRIVATE ${FFI_LIBRARY_PATH} )
+  target_link_libraries( LLVMInterpreter PRIVATE ffi )
 endif()
-- 
2.33.1

